substrate/frame/treasury/src/lib.rs

```diff
@@ -453,17 +465,29 @@ pub mod pallet {
 			}
 
 			// Check to see if we should spend some funds!
-			if (n % T::SpendPeriod::get()).is_zero() {
-				Self::spend_funds()
+			let last_spend_period = LastSpendPeriod::<T, I>::get()
+				// This unwrap should only occur one time on any blockchain.
+				// `update_last_spend_period` will populate the `LastSpendPeriod` storage if it is
+				// empty.
+				.unwrap_or(Self::update_last_spend_period());
```

Comment:
It is possible to use the [`OnEmpty` syntax](https://github.com/paritytech/polkadot-sdk/blob/cd91c6b782b60906597ed28c4e4a4fdddd8428c9/substrate/frame/examples/kitchensink/src/lib.rs#L130-L137) of the `LastSpendPeriod` and then change it to `ValueQuery` - it is not really optional.

Reply:
Feel like it is exactly the same thing, but more substrate magic.

or what am I missing?

Reply:
Yup. Depends on whether you like more or less magic 😆